README-3PP-LICENSE-CHECK-AND-AUDIT.md

```diff
@@ -0,0 +1,121 @@
+# supply your own token. See: https://github.com/eclipse/dash-licenses#automatic-ip-team-review-requests
+export DASH_LICENSES_TOKEN=<token>
+
+# Run dash-licenses on each repo - a summary.txt file will be created, 
+# that contains the results
+find vscode/extensions -maxdepth 1 -mindepth 1 -type d -exec bash -c "cd '{}' &&  pwd && java -jar $DASH_LICENSES_JAR yarn.lock -timeout 120 -batch 20 -summary ./summary.txt" \;
+
+# gather all output files and filter for restricted 3PPs:
+cat vscode/extensions/*/summary.txt | grep restricted | sort | uniq
+```
+
+```bash
+# (optional) dash-licenses with Automatic IP Team Review Requests
+
+# You need to supply your own Eclipse Foundation GitLab token. See: 
+# https://github.com/eclipse/dash-licenses#automatic-ip-team-review-requests
+export DASH_LICENSES_TOKEN=<token>
+
+# run dash-licenses in review mode, that automatically submits suspicious 
+# dependencies for review, by the Eclipse Foundation IP team.
+find vscode/extensions -maxdepth 1 -mindepth 1 -type d -exec bash -c "cd '{}' &&  pwd && java -jar $DASH_LICENSES_JAR yarn.lock -timeout 120 -batch 20 -summary ./summary.txt -review -token $DASH_LICENSES_TOKEN -project ecd.theia" \;
+
+```
+
+### Security vulnerability audit (builtins)
+
+The vscode repo uses `yarn`, so we can use `yarn audit` on the individual extensions directories, that each contain a small `yarn` sub-project (at least those that have code). We are mostly interested in runtime vulnerabilities (vs dev and test) of level `hight` and up.
+
+```bash
+find vscode/extensions -maxdepth 1 -type d \( ! -name . \) -exec bash -c "cd '{}' && pwd && yarn audit --level high --groups dependencies" \;
+```
+
+## External builtin vscode extensions
+
+`External builtin vscode extensions` are extensions that are bundled with vscode but not built along with it. Instead they are fetched from the `Visual Studio Marketplace`, when vscode is built. For Theia, some are published to [OpenVSX](https://open-vsx.org/) by the [openvsx bot](https://github.com/open-vsx/publish-extensions) and some may not be available (e.g. that are compatible with `vscode API v1.55.2`).
+
+They are defined in `vscode`'s root `product.json` - each one at the version that needs to be used, for that vscode baseline. The exact list of extensions changes depending on `vscode` baseline.
+
+For convenience, we added an npm script that clones these external builtin extensions repositories and checks-out each one at the expected version (as per `product.json` in current `vscode` git submodule baseline):
+
+```bash
+# check-out a given vscode tag. e.g. 1.55.2:
+cd vscode && git checkout 1.55.2 && cd ..
+# 'yarn` is only needed to install the repo's dependencies, if not already done. You do 
+# not need to let it complete the builtin extensions build, for the current purpose
+yarn
+# obtain external builtins and check-out each one to the "correct" version. They will be 
+# stored in subfolder "external-builtins"
```

Comment:
I did that on purpose so that we do not need to check for its existence or create the directory. `external-builtin-extensions-here.txt` is a place-holder, that can be removed if we eventually have real content under that directory. (at first this prodecure was in there, but I moved it up to the root when I generalized it to both internal and external builtins).

Does it make sense?

Reply:
update: I have done as suggested below and removed this file and folder from source control.